libr/debug/debug.c

```diff
@@ -1359,7 +1359,8 @@ R_API int r_debug_child_clone(RDebug *dbg) {
 }
 
 R_API bool r_debug_is_dead(RDebug *dbg) {
-	bool is_dead = (dbg->pid == -1 && strncmp (dbg->h->name, "gdb", 3));
+	bool is_dead = (dbg->pid == -1 && strncmp (dbg->h->name, "gdb", 3))
+		|| (dbg->reason.type == R_DEBUG_REASON_DEAD);
```

Comment:
I made a PR fixing this issue, the only problem is reset dbg->reason.type when reload binary.

Ur fix work if u run a fresh r2 debug session, but dont work when r2 reload a debuggee.